presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -198,6 +200,64 @@ public static Slice stCentroid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return serialize(createFromEsriGeometry(centroid, geometry.getEsriSpatialReference()));
     }
 
+    @Description("Returns the minimum convex geometry that encloses all input geometries")
+    @ScalarFunction("ST_ConvexHull")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stConvexHull(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (!isValidGeometryForConvexHull(geometry)) {
+            return serialize(geometry);
+        }
+        OGCGeometry convexHull = geometry.convexHull();
+        long pointCount = getOGCGeometryPointCount(convexHull);
+        if (pointCount == 0) {
+            return serialize(getSinglePointFromGeometry(geometry));
+        }
+        else if (pointCount == 2) {
```

Comment:
This looks like a bug in the ESRI's library. I submitted an issue: https://github.com/Esri/geometry-api-java/issues/172

Thanks for coming up with a workaround. 

tranformOGCGeometryToOGCLineString function is very short and is used only once. Let's inline it here.